src/decimal.rs

```diff
@@ -0,0 +1,32 @@
+#![allow(unused_variables)]
+#![doc(cfg(feature = "decimal"))]
+use rust_decimal::Decimal;
+use leptos::*;
+#[derive(Default)]
+pub struct DecimalNumberRenderOptions {
```

Comment:
We could simply re-use `NumberRenderOptions`, right?

Reply:
We totaly _could_, my point to have this its own type is that i could imagine these evolving separatly. 

I have no real example at the top of my head right now where these would be different in the future, and i guess we could split it up if that was to happen. Let me know what you think.